makeReducedConfig.py

```diff
@@ -0,0 +1,209 @@
+        ref_inputs = ref_node.getInputConnections()
+        if verbose: print "\tInputs:", ["%s -> %s"%(r.cinput.name, r.coutput.name) for r in ref_inputs]
+        for ref_c in ref_inputs:
+            ref_n = ref_c.cinput
+            if self.isIncluded(ref_n):
+                if verbose: print "\t\t", "Found good input!:", ref_n.name
+                in_n = node(ref_n.name)
+                in_c = connection(ref_c.memory, in_n, n, ref_c.cinext, ref_c.coutext)
+                n.addConnection(in_c)
+                in_n.addConnection(in_c)
+                self.addConnection(in_c)
+                if in_n.name != "" and in_n.name not in self.nodes:
+                    self.addNode(in_n)
+                    self.findInputConnections(in_n, ref_p)
+
+    def findOutputConnections(self, n, ref_p):
+        if verbose: print "\t", n.name
+        ref_node = ref_p.nodes[n.name]
+        ref_outputs = ref_node.getOutputConnections()
+        if verbose: print "\tOutputs:", ["%s -> %s"%(r.cinput.name, r.coutput.name) for r in ref_outputs]
+        for ref_c in ref_outputs:
+            ref_n = ref_c.coutput
+            if self.isIncluded(ref_n):
+                if verbose: print "\t\t", "Found good output!:", ref_n.name
+                out_n = node(ref_n.name)
+                out_c = connection(ref_c.memory, n, out_n, ref_c.cinext, ref_c.coutext)
+                n.addConnection(out_c)
+                out_n.addConnection(out_c)
+                self.addConnection(out_c)
+                if out_n.name != "" and out_n.name not in self.nodes:
+                    self.addNode(out_n)
+                    self.findOutputConnections(out_n, ref_p)
+
+    def isIncluded(self, n):
+        if n.name.startswith("TC_L1L2"):
+            for tc in self.tcs:
+                if n.name == "TC_L1L2%s"%tc: return True
+            return False
+        if n.name.startswith("TE_L1PHI"):
+            for l1 in self.l1phis:
+                for lx in self.lxphis:
+                    if "L1PHI%s"%l1 in n.name and "L2PHI%s"%lx in n.name: return True
+            return False
+        if "L1PHI" in n.name:
+            for l1 in self.l1phis:
+                if "L1PHI%s"%l1 in n.name: return True
+            return False
+        if "PHI" in n.name:
+            for lx in self.lxphis:
+                for x in [2,3,4,5,6]:
+                    if "L%sPHI%s"%(x,lx) in n.name: return True
+            return False
+        if n.name in ["FT_L1L2", "PD", ""]: return True
+        return False
+
+# ----------------------------------------------------
+# Main function
+# ----------------------------------------------------
+
+# Load in full project
+print "Loading full wire project..."
+full_wires = project()
+full_wires.loadProject("wires.dat")
+full_wires.saveProject("test.dat")
+
+# Set up reduced project and give it a phi sector in L1
+print "Finding reduced configuration..."
+reduced_wires = project()
+reduced_wires.addTC("F", full_wires)
```

Comment:
If this says create chain for phi region "F", please make the phi region an argument that is passed to the script, with default value F.

Reply:
Done